com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoopDef.hlsl

```diff
@@ -9,6 +9,11 @@
 
 #define DWORD_PER_TILE 16 // See dwordsPerTile in LightLoop.cs, we have roomm for 31 lights and a number of light value all store on 16 bit (ushort)
 
+// Some file may not required HD shadow context at all. In this case provide an empty one
+#ifndef HAVE_HD_SHADOW_CONTEXT
+struct HDShadowContext { float unused; };
```

Comment:
What is the purpose of this? The definition of shadow context will mismatch whatever is going to be used by the functions, so it won't compile anyway?

Reply:
HDShadowContext  is only define in HDShadowContext. (where I now define HAVE_HD_SHADOW_CONTEXT). Several files required to include LightloopDefs that require a shadow context so all those files include hdshadow.hlsl implicitely via lighting.hlsl whereas they don't need all those files. To remove those around 8 files dependency just to declare a HDShadowContext   that is not used, we do this.

Reply:
I see, though unless I am missing something this now creates a dependency on the order of include so I guess that would need to be made more explicit in comments. 

Say I include HDShadowContext.hlsl _after_ I include LightloopDef , I will double define the struct which I assume will not compile. 
Not sure if that dependency already exists, but I can see it creating some potentially tricky issues

Reply:
problem is already there, you can't do it the other way :)

Reply:
Sure, but before it would complain about a struct not being defined (so it is obvious you are missing an include), now it will complain about it being defined twice which is a bit less obvious. I would add a small comment in here that if the double defintion error occur is likely because you need to include HDShadowContext.hlsl beforehand

Reply:
I added a comment. 

> but before it would complain about a struct not being defined
if you do wrong order guess there is more trouble than this one :), but a comment never hurt